src/index.js

```diff
@@ -0,0 +1,141 @@
+    event.target.completed = true;
+  } else if (event.target.type === 'checkbox' && !event.target.checked) {
+    event.target.nextElementSibling.style.textDecoration = 'none';
+  }
+});
+
+if (localStorage.getItem('listArray')) {
+  listArray = JSON.parse(localStorage.getItem('listArray'));
+  UI.displayData();
+}
+
+listHolder.addEventListener('click', (event) => {
+  if (event.target.classList.contains('fa-ellipsis-v')) {
+    event.target.classList.remove('fa-ellipsis-v');
+    event.target.classList.add('fa-trash');
+    event.target.parentElement.parentElement.style.backgroundColor = '#f39c12';
+    const targeted = event.target.parentNode.previousElementSibling.lastElementChild;
+    targeted.contentEditable = 'true';
+  } else if (event.target.classList.contains('fa-trash')) {
+    const index = event.target.parentElement.parentElement.getAttribute('data-index');
+    listArray.splice(index, 1);
+    localStorage.setItem('listArray', JSON.stringify(listArray));
+    event.target.parentElement.parentElement.remove();
+  }
+});
+listHolder.addEventListener('input', (e) => {
+  if (e.target.closest('#discription')) {
+    const inputText = e.target.closest('#discription');
+    const taskElm = inputText.parentNode.parentNode;
+    const index = Array.prototype.indexOf.call(listHolder.children, taskElm);
+    listArray[index].discription = e.target.textContent;
+
+    localStorage.setItem('listArray', JSON.stringify(listArray));
+  }
+});
+
+window.addEventListener('load', () => {
+  if (localStorage.getItem('listArray')) {
+    listArray = JSON.parse(localStorage.getItem('listArray'));
+    UI.displayData();
+  }
+});
+/*
+lists.forEach((list) =>{
+   list.addEventListener('click', () => {
+    console.log('clicked');
+   })
+})
+
+  Editing the list *********inputText.addEventListener(
+      'input',
+      (e) => {
+        const parent = document.querySelector('.list-container');
+        const child = e.target.closest('.task-container');
+        saveData(editText(loadData(), parent, child, inputText.textContent));
+      },
+      false,
+    );
+*/
+
+if (localStorage.getItem('listArray')) {
+  listArray = JSON.parse(localStorage.getItem('listArray'));
+  UI.displayData();
+}
+
+// Function to remove completed items from the listArray and the DOM
+function removeCompleted() {
+  listArray = listArray.filter((item) => !item.completed);
+  localStorage.setItem('listArray', JSON.stringify(listArray));
+  UI.displayData();
+  // Select all elements with the completed class and remove them from the DOM
+  const completedElements = document.querySelectorAll('.completed');
+  completedElements.forEach((element) => element.remove());
+}
+
+// Add a click event listener to the cleanAllDone element
+cleanAllDone.addEventListener('click', removeCompleted);
+
+if (localStorage.getItem('listArray')) {
+  listArray = JSON.parse(localStorage.getItem('listArray'));
+  UI.displayData();
+}
```

Comment:
- [ ] As requested by the previous reviewer, kindly create a new JavaScript file for the new functionality for adding, deleting, and editing the description of a task.